app/frontend/packs/components/checkbox_search_filter.js

```diff
@@ -0,0 +1,202 @@
+  }
+
+  sanitizeText (text) {
+    text = text.replace(/&/g, 'and')
+    text = text.replace(/[’',:–-]/g, '') // remove punctuation characters
+    text = text.replace(/[.*+?^${}()|[\]\\]/g, '\\$&') // escape special characters
+    return text.trim().replace(/\s\s+/g, ' ').toLowerCase() // replace multiple spaces with one
+  }
+
+  setContainerHeight (height) {
+    this.checkboxesContainer.style.height = height
+  }
+
+  setup () {
+    this.setupStatusBox()
+    this.setupHeading()
+    if (this.checkboxes.length >= 15) {
+      this.setupTextBox()
+    }
+    this.setupHeight()
+  }
+
+  setupHeading () {
+    const element = document.createElement('p')
+    element.className = 'app-checkbox-filter__title'
+    element.setAttribute('aria-hidden', 'true')
+
+    this.heading = element
+    this.container.prepend(this.heading)
+  }
+
+  setupHeight () {
+    let initialOptionContainerHeight = this.checkboxesContainer.style.height
+    let height = outerHeight(this.checkboxesInnerContainer)
+
+    // check whether this is hidden by progressive disclosure,
+    // because height calculations won't work
+    if (this.checkboxesContainer.offsetParent === null) {
+      initialOptionContainerHeight = 200
+      height = 200
+    }
+
+    // Resize if the list is only slightly bigger than its container
+    if (height < initialOptionContainerHeight + 50) {
+      this.setContainerHeight(height + 1)
+      return
+    }
+
+    // Resize to cut last item cleanly in half
+    const visibleCheckboxes = this.getVisibleCheckboxes()
+    const lastVisibleCheckbox = visibleCheckboxes[visibleCheckboxes.length - 1]
+    const position = lastVisibleCheckbox.parentNode.offsetTop // parent element is relative
+    this.setContainerHeight(position + (lastVisibleCheckbox.height / 1.5))
+  }
+
+  setupStatusBox () {
+    const element = document.createElement('div')
+    element.className = 'govuk-visually-hidden'
+    element.id = this.container.id + '-checkboxes-status'
+    element.setAttribute('role', 'status')
+
+    this.statusBox = element
+    this.updateStatusBox({
+      foundCount: this.getAllVisibleCheckboxes().length,
+      checkedCount: this.getAllVisibleCheckedCheckboxes().length
+    })
+    this.container.insertAdjacentElement('afterend', this.statusBox)
+  }
+
+  setupTextBox () {
+    const tagContainer = this.container.querySelector('.app-checkbox-filter__selected')
+    const textBoxElements = this.getTextBoxHtml()
+
+    if (tagContainer) {
+      for (let i = 0; i < textBoxElements.length; i++) {
+        tagContainer.parentNode.insertBefore(textBoxElements[i], tagContainer.nextSibling)
+      }
+    } else {
+      for (let i = 0; i < textBoxElements.length; i++) {
+        this.heading.parentNode.insertBefore(textBoxElements[i], this.heading.nextSibling)
+      }
+    }
+
+    this.textBox = this.container.querySelector('.app-checkbox-filter__filter-input')
+    this.textBox.addEventListener('keyup', () => this.onTextBoxKeyUp(this))
```

Comment:
Does this get polyfilled? Won't work in IE8, for example.
https://caniuse.com/?search=addEventListener

Reply:
This one does: https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/vendor/polyfills/Event.js